sbt_common/messaging/src/main/scala/uk/ac/wellcome/messaging/message/MessageWriter.scala

```diff
@@ -39,19 +40,43 @@ class MessageWriter[T] @Inject()(
     s"$topicName/${dateFormat.format(currentTime)}/${currentTime.getTime.toString}"
   }
 
-  def write(message: T, subject: String): Future[PublishAttempt] = {
+  def write(message: T, subject: String)(
+    implicit encoder: Encoder[T],
+    notificationEncoder: Encoder[MessageNotification[T]])
+    : Future[PublishAttempt] =
     for {
-      location <- objectStore.put(messageConfig.s3Config.bucketName)(
-        message,
-        keyPrefix = KeyPrefix(getKeyPrefix())
-      )
-      _ = debug(s"Successfully stored message $message in location: $location")
+      encodedString <- Future.fromTry(toJson(message))
+
+      // If the encoded message is less than 250KB, we can send it inline
+      // in SNS/SQS (although the limit is 256KB, there's a bit of overhead
+      // caused by the notification wrapper, so we're conservative).
+      //
+      // Max SQS message size:
+      // https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-limits.html#limits-messages
+      //
+      // Max SNS message size:
+      // https://aws.amazon.com/sns/faqs/
+      //
+      notification: MessageNotification[T] <- if (encodedString.length > 250 * 1000) {
+        writeToS3(encodedString)
+      } else {
+        Future.successful(InlineNotification(message))
+      }
+
```

Comment:
This relies on the SNSWriter using the same JSON serialization as `encodedString <- Future.fromTry(toJson(message))`.  Would it be possible to build the whole notification as a string before the test and then use the `writeMessage(message: String, subject: String)` method?

This still assumes the underlying byte serialisation used by AmazonSNS, which we could make a bit more explicit `string.getBytes("UTF-8").length`.